src/urllib3/util/response.py

```diff
@@ -53,43 +53,40 @@ def assert_header_parsing(headers):
     if not isinstance(headers, httplib.HTTPMessage):
         raise TypeError(f"expected httplib.Message, got {type(headers)}.")
 
-    defects = getattr(headers, "defects", None)
-    get_payload = getattr(headers, "get_payload", None)
-
     unparsed_data = None
-    if get_payload:
-        # get_payload is actually email.message.Message.get_payload;
-        # we're only interested in the result if it's not a multipart message
-        if not headers.is_multipart():
-            payload = get_payload()
-
-            if isinstance(payload, (bytes, str)):
-                unparsed_data = payload
-    if defects:
-        # httplib is assuming a response body is available
-        # when parsing headers even when httplib only sends
-        # header data to parse_headers() This results in
-        # defects on multipart responses in particular.
-        # See: https://github.com/urllib3/urllib3/issues/800
-
-        # So we ignore the following defects:
-        # - StartBoundaryNotFoundDefect:
-        #     The claimed start boundary was never found.
-        # - MultipartInvariantViolationDefect:
-        #     A message claimed to be a multipart but no subparts were found.
-        defects = [
-            defect
-            for defect in defects
-            if not isinstance(
-                defect, (StartBoundaryNotFoundDefect, MultipartInvariantViolationDefect)
-            )
-        ]
+
```

Comment:
I figured since we're going to start protecting this with typechecking we could remove the runtime check on if attributes exist. Can revert though.

Reply:
This is all fine since we have the `isinstance` check, so we don't have to worry about users sending weird objects into this function